app/controllers/assessments_controller.rb

```diff
@@ -137,7 +137,6 @@ def update
     redirect_to(root_url) && return if patient.nil?
 
     redirect_to root_url unless current_user&.can_edit_patient_assessments?
-    patient = Patient.find_by(submission_token: params.permit(:patient_submission_token)[:patient_submission_token])
```

Comment:
Something seems wrong here in general. This line: 
`submission_token = get_newest_submission_token(params.permit(:patient_submission_token)[:patient_submission_token])`
is assuming the `patient_submission_token` param is the old token always. How is it even handling the new tokens? @timwongj . Also the `Patient.find_by` in `get_newest_submission_token` should filter out purged.

Reply:
The reason I removed this line was that the patient was already found on L136 through use of the `get_newest_submission_token` helper and this line I deleted seemed to be assuming that the token would always be the new token format.